### PR TITLE
Custom CSS classes and mobile image attribute

### PR DESCRIPTION
**Pull request by salesfelipe**
#### What is the purpose of this pull request?
Extends the carousel component adding the mobile image attribute and adds specific CSS classes to the subcomponents(arrow, dots).

#### What problem is this solving?
Displays another image when the user is on a mobile device and give to the developers that are using this component the ability to customize its CSS using its own specific classes. 

#### How should this be manually tested?

https://carousel--storecomponents.myvtex.com/test

#### Screenshots or example usage

image

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/Banner.js [hunk 1]
```diff
@@ -7,57 +7,51 @@ import { Link } from 'render'
...
+              width="100%"
+              className="img-regular"
+              src={image}
+              alt={description}
+            />
+            <div
+              className="img-mobile"
+              style={{ maxHeight: `${mobileHeight}px` }}
+            >
+              <img width="100%" src={mobileImage} alt={description} />
```
Comment (ClaudivanFilho) [inline react/Banner.js]: tachyons

File: react/Banner.js [hunk 2]
```diff
@@ -7,57 +7,51 @@ import { Link } from 'render'
...
+      mobileHeight,
+    } = this.props
 
     return (
       <div>
         <Link page={page} params={targetParams}>
-          <img width="100%" src={image} alt={description}/>
+          <div className="img-container">
+            <img
+              width="100%"
```
Comment (ClaudivanFilho) [inline react/Banner.js]: tachyons

File: react/Banner.js [hunk 3]
```diff
@@ -7,57 +7,51 @@ import { Link } from 'render'
...
+  mobileImage: PropTypes.string.isRequired,
+  /** Max height size of the banner on mobile  */
+  mobileHeight: PropTypes.number.isRequired,
   /** The description of the image */
   description: PropTypes.string.isRequired,
   /** The page where the image is pointing to */
   page: PropTypes.string,
   /** Params of the url */
-  targetParams: PropTypes.object
+  targetParams: PropTypes.object,
```
Comment (ClaudivanFilho) [inline react/Banner.js]: tem como destrinchar mais esse object? colocar um PropTypes.shape

File: pages/pages.json [hunk 4]
```diff
@@ -1,34 +1,14 @@
 {
   "pages": {
     "carousel": {
-      "path": "/",
+      "path": "/test",
```
Comment (leticiawanderley) [inline pages/pages.json]: Not sure about that, but I have no idea how it was supposed to be.

File: react/Banner.js [hunk 5]
```diff
@@ -7,57 +7,51 @@ import { Link } from 'render'
...
-        },
-      },
-    },
-  },
-}
-
 Banner.propTypes = {
   /** The image of the banner */
   image: PropTypes.string.isRequired,
+  /** The image of the banner */
```
Comment (leticiawanderley) [inline react/Banner.js]: Maybe add "on mobile"

File: react/Banner.js (hunk 2 above)
Comment (leticiawanderley) [inline react/Banner.js]: Have you tried using tachyons class `w-100`?

File: react/Carousel.js [hunk 6]
```diff
@@ -51,22 +45,33 @@ class Carousel extends Component {
...
     return (
-      <div className="vtex-carousel pa7">
+      <div className="vtex-carousel">
         {!loading && (
           <Slider {...settings}>
             {banners.map(function(banner, i) {
               if (banner && banner.image) {
                 return (
-                  <div key={i}>
+                  <div key={i} style={{ maxHeight: `${height}px` }}>
```
Comment (leticiawanderley) [inline react/Carousel.js]: Use something else as key.
https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318

File: pages/pages.json (hunk 4 above)
Comment (salesfelipe) [reply pages/pages.json]: 👍

File: react/Banner.js (hunk 5 above)
Comment (salesfelipe) [reply react/Banner.js]: 👍

File: react/Banner.js (hunk 2 above)
Comment (salesfelipe) [reply react/Banner.js]: 👍

Comment (salesfelipe) [reply react/Banner.js]: 👍

File: pages/pages.json (hunk 4 above)
Comment (brunojdo) [reply pages/pages.json]: Who knows @leticiawanderley ? @klzns ?

File: react/Arrow.js [hunk 7]
```diff
@@ -25,8 +25,8 @@ Arrow.propTypes = {
   style: PropTypes.object,
   /** Maximum number of items in the shelf. */
   onClick: PropTypes.func,
-  /** The color of the arrow icon. Ex: '#F00', 'rgb(255, 0, 0)'. */
-  color: PropTypes.string.isRequired
+  /** The css class that specifies the arrow */
```
Comment (brunojdo) [inline react/Arrow.js]: remove "The"

File: react/Carousel.js [hunk 8]
```diff
@@ -242,37 +381,25 @@ Carousel.propTypes = {
   autoplay: PropTypes.bool.isRequired,
   /** How long it should wait to change the banner in secs */
   autoplaySpeed: PropTypes.number,
+  /** Max height size of the banners */
+  height: PropTypes.number.isRequired,
+  /** Max height size of the banners on mobile */
+  mobileHeight: PropTypes.number.isRequired,
   /** Should show the dots or not */
```
Comment (brunojdo) [inline react/Carousel.js]: /** Set visibility of dots */

File: react/Carousel.js [hunk 9]
```diff
@@ -242,37 +381,25 @@ Carousel.propTypes = {
...
   showDots: PropTypes.bool,
   /** Should show the arrows or not */
```
Comment (brunojdo) [inline react/Carousel.js]: /** Set visibility of arrows */

File: react/Dots.js [hunk 10]
```diff
@@ -9,8 +9,8 @@ class Dots extends Component {
     const { className, style, dots, color } = this.props
 
     return (
-      <div className={className} >
-        <ul className="ma0 pa0" style={{ ...style, color: color || '#000' }}>
+      <div className={`vtex-carousel__dots ${className}`}>
+        <ul className="ma0 pa0" style={{ ...style }}>
```
Comment (brunojdo) [inline react/Dots.js]: We don't need a namespace here? CC: @leticiawanderley @klzns

File: react/Arrow.js (hunk 7 above)
Comment (salesfelipe) [reply react/Arrow.js]: 👍

File: react/Carousel.js (hunk 8 above)
Comment (salesfelipe) [reply react/Carousel.js]: 👍

File: react/Carousel.js (hunk 9 above)
Comment (salesfelipe) [reply react/Carousel.js]: 👍

**Pull request merged**